percona/controllers/template/template.go

```diff
@@ -103,6 +115,43 @@ func GetPMMContainerJSON(pgc *crv1.PerconaPGCluster, nodeName string) ([]byte, e
 	return b, nil
 }
 
+func handleAffinityTemplate(template []byte, affinity crv1.PodAffinity, withNodeSelector bool) ([]byte, error) {
+	affinityBytesWithNodeSelector := []byte(`{
+		{{if .NodeSelector}}
+		"nodeAffinity": {{ .NodeSelector }}
+		{{ end }}
+		{{if and .NodeSelector .PodAntiAffinity}},{{end}}
+		{{.PodAntiAffinity}}
+	  }`)
+	affinityBytes := []byte(`{
+		{{.PodAntiAffinity}}
+				}`)
```

Comment:
nitpick but these lines really look ugly. how about?

```suggestion
	affinityBytesWithNodeSelector := []byte(`{
		{{if .NodeSelector}}
		"nodeAffinity": {{ .NodeSelector }}
		{{ end }}
		{{if and .NodeSelector .PodAntiAffinity}},{{end}}
		{{.PodAntiAffinity}}
	}`)
	affinityBytes := []byte("{ {{.PodAntiAffinity}} }")
```